rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -397,8 +434,29 @@ def _decode_transaction(
         counterparties = set()
         refresh_balances = False
 
+        # Check if any rules should run due to the 4bytes signature of the input data
+        fourbytes = transaction.input_data[:4]
+        input_data_rules = self.rules.input_data_rules.get(fourbytes)
+
         # decode transaction logs from the receipt
         for tx_log in tx_receipt.logs:
+            if input_data_rules and len(tx_log.topics) != 0 and (input_rule := input_data_rules.get(tx_log.topics[0])) is not None:  # noqa: E501
+                context = DecoderContext(
+                    tx_log=tx_log,
+                    transaction=transaction,
+                    decoded_events=events,
+                    all_logs=tx_receipt.logs,
+                    action_items=action_items,
+                )
+                try:  # run specific decoder if the 4bytes signature + topic match
+                    result = input_rule(context)
+                except (DeserializationError, ConversionError, UnknownAsset) as e:
+                    log.error(f'Decoding log {tx_log} of {transaction} failed due to {e!s}')
+                    result = DEFAULT_DECODING_OUTPUT
+
+                if result.event:
+                    events.append(result.event)
```

Comment:
should thsi do a continue? otherwise the decoding by address logic is executed

Reply:
hmm yes I think it should continue here